### DISCUSSION TITLE
I'm lost on how to build a remote.txt for my cable box

### DISCUSSION THREAD

**Discussion by shoobydoodoo** [GerryDazoo/Slinger#563] [Q&A]
Please excuse my computer illiteracy, but after reading the release notes and searching on Github discussion, I'm still not sure how to make a custom remote.txt for my cable box in Taiwan.

I'm using two Pro-HDs and a Solo

In release note V3.0x, the line, "Make sure the remote is working with the Slingplayer/Slingplayer for Web software." So for all the instructions that follow about matching the correct numbers to their respective buttons in remote.txt., this is under the condition that the remote.bin file had already been "loaded" into the Slingbox in the first place. Correct? I used to use web Slingplayer to upload the bin files, but now that it no longer works, do we use a legacy desktop Slingplayer to upload the bin files? If so which version should I use for Pro-HD and Solo? I'm using Windows 10 and Mac Big Sur.

For my video inputs that had bin files configured before N9, the remote seems to work fine even when I put a ";" before "remote include=" as long as I have the correct remote code. For the tuner, I need to rely on the ProHD_Tuner_.txt. Now there's a component input on the Solo that I need to setup. The 5678 port has been blocked and the video streams fine, but I have trouble figuring out how to make the remote work. I have the old bin.file.

In this [Github discussion](https://github.com/GerryDazoo/Slinger/discussions/490), I'm not sure if OP and I are asking the same question. Gerry seemed to link him to instruction on how to build remote file from scratch. Is that for the bin file?

Any help or pointer is appreciated. Thanks!

**Comment by regmar505** [comment]
I have the same issue here. I carried my Slingbox M2 to a location where the Set Top Box is different therefore the remote control which is loaded in the Slingbox does not match the current STB. I tried almost all custom remotes and I cannot get anywhere. An answer to the previous post would be greatly appreciated.

**Reply by shoobydoodoo** [reply]
> I have the same issue here. I carried my Slingbox M2 to a location where the Set Top Box is different therefore the remote control which is loaded in the Slingbox does not match the current STB. I tried almost all custom remotes and I cannot get anywhere. An answer to the previous post would be greatly appreciated.

Hi, I'm not sure how to setup remote control for M2, as it's the latest Slingbox and seems to use different software. Some nice person over at [Reddit](https://www.reddit.com/r/slingbox/comments/1207riw/im_lost_on_how_to_build_a_remotetxt_for_my_cable/) helped me figure out how to upload the bin file using legacy 1.5x Slingplayer, which doesn't seem to be compatible with M2. I'd be very interested to know how to do it for the later models as well though. Hopefully someone can help, or if you figure it out, please reply here. Thanks and good luck!

**Reply by mike-uk** [reply]
I wonder how to do that as well, as I want to setup a set top box with Pro-HD in asia. I wonder if it would be easier just to get a smart network IR blaster that you can use an app to program what to send. People seem to link that up to switch on air con with their phone just before they get home.

I do know you can use arduino to read the hex code for each button but I can't see how you can convert that to bin. Maybe there is some way to capture the raw ir then enter into remote master software to make the bin.

I understand now, once you have the bin for the remote you have to load it to solo or pro-hd with slingplayer 1.5 then match the key with remote.txt

**Reply by mike-uk** [reply]
I found you can use arduino nano and QSE159 photo sensor to make an ir widget to capture remote to be used in RemoteMaster to create a .bin file. Then .bin load to Pro-HD with Slinglayer 1.5 and then map the keys in slinger with remote.txt

Airwidget
http://www.hifi-remote.com/forums/viewtopic.php?p=145548&sid=717f01350094b6e7a3de6f3bde782d63

How to create a custom remote guide (this is the same one information but all the links work)
http://www.hifi-remote.com/wiki/index.php?title=How_to_Create_Custom_Remote_Controls_and_use_them_on_your_Slingbox

**Reply by shoobydoodoo** [reply]
Hey great find! That looks like it involves some learning curve and hardware investment. I'd be up for it if I had the time, but currently it seems the easiest and fastest way is to buy a broadlink ir blaster like you mentioned. This [Github thread](https://github.com/GerryDazoo/Slinger/discussions/398) discusses some cheap solutions. 

If you prefer to get a bin file to get the native Slingbox IR to work, perhaps you can ask on the hifi-remote forum if anyone is nice enough to build a bin file for you. That's how I got my custom bin files many years ago. I remember there are a lot of nice people on there, which I was grateful for. Good luck!

**Reply by regmar505** [reply]
@shoobydoodoo Thanks for sharing the Github link. I believe the Broadlink RM4 is the way to go for a Slingbox M2. I am worried to reset my M2 as I did not read any success story so far about this model.

**Reply by mike-uk** [reply]
@shoobydoodoo  Yeah its a deep learning curve and no good for newer slingbox that need. It is not that expensive a cloned arduino nano on aliexpress is less than $3usd. I actually have most of the parts already apart from the recommended it sensor. I am actually learning to program so all this stuff interests me as well.

@regmar505 I see they have a mini version of Broadlink RM4 now which is only half the price on aliexpress, it seem a good replacement and easier to use. I saw the demo video you use an app. How do you program the keys into the app?
